File: app/src/main/java/com/falcon/restaurants/network/RetrofitInterface.java
package com.falcon.restaurants.network;

import com.falcon.restaurants.network.restaurant.RestaurantNet;
import com.falcon.restaurants.network.meal.MealNet;
import java.util.List;
import io.reactivex.Single;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface RetrofitInterface {

    @GET("api/getRestaurants.php")
    public Single<List<RestaurantNet>> getRestaurants(@Query("maxUpdatedAt") String maxUpdatedAt);

    @GET("api/getMeals.php")
    public Single<List<MealNet>> getMeals(@Query("maxUpdatedAt") String maxUpdatedAt);

    @FormUrlEncoded
    @POST("cpanel/login.php")
    public Single<List<MealNet>> login(@Field("name") String name, @Field("password") String password);

    @FormUrlEncoded
    @POST("cpanel/register.php")
    public Single<List<MealNet>> register(@Field("name") String name, @Field("email") String email, @Field("password") String password);

}

File: app/src/main/java/com/falcon/restaurants/data/network/RetrofitInterface.kt
package com.falcon.restaurants.data.network
import com.falcon.restaurants.data.network.restaurant.RestaurantNet
import com.falcon.restaurants.data.network.meal.MealNet
import io.reactivex.Single
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface RetrofitInterface {

    @GET("api/getRestaurants.php")
    fun  getRestaurants(@Query("maxUpdatedAt") maxUpdatedAt: String ): Single<List<RestaurantNet>>

    @GET("api/getMeals.php")
    fun getMeals(@Query("maxUpdatedAt") maxUpdatedAt: String): Single<List<MealNet>>

    @FormUrlEncoded
    @POST("api/login.php")
    fun login(@Field("name") name: String, @Field("password") password: String): Single<List<MealNet>>

    @FormUrlEncoded
    @POST("api/register.php")
    fun register(@Field("name") name: String,
                 @Field("email") email: String,
                 @Field("password") password: String): Single<List<MealNet>> 

}
